Guide/DynamicProgramming/README.md

```diff
@@ -0,0 +1,12 @@
+# Dynamic Programming Guide
+
+## Overview
+<p> Dynamic Programming - An algorithm that solves smaller problems that can
```

Comment:
What about this?

Dynamic Programming: An algorithm that solves a problem by breaking it into smaller sub-problems, for which the solutions of the sub-problems can be combined to solve the original problem. @NilsG-S that sound better?

[Memoization](http://stackoverflow.com/questions/1988804/what-is-memoization-and-how-can-i-use-it-in-python) is typically used to solve sub-problems and combine them to solve the original problem without using recursion, a more computationally intensive method.